src/utils/system-generator/generate-sector.ts

```diff
@@ -16,14 +16,21 @@ import {
 } from './system-functions';
 import { systemNameGenerator } from './system-name-generator';
 import { Sector } from '../../redux/sectorSlice';
+import { Ship, ship1 } from './ship';
 
 export interface System {
   systemStar: string;
   systemPlanets: Array<Planet>;
   systemName: string;
   cords: string;
   ownership: string;
-  hangar: string[];
+  hangar: Ship[];
+}
+
+export const buildShip = (system:System) => {
+  console.log('help')
+      setTimeout(function(){system.hangar.push(ship1); console.log(system.hangar)}, ship1.buildTime*1000)
```

Comment:
Hi Shirley, I'm not seeing the ship added to the system's array when I click this button. It's throwing this error `generate-sector.ts:32 Uncaught TypeError: Cannot add property 0, object is not extensible`

Reply:
P.S the UI will update if there's something in the hangar. Or at least it SHOULD, we haven't tested that yet

Reply:
Look into Object.freeze and checking to find out when the array/object in question is frozen or not. It might have something to do with set time out having some sort of side effect on the object frozen status.

Reply:
@DylanBozarth @Sjazfrey Do we know if this is ready to merge?